src/layers/generic.cpp

```diff
@@ -221,9 +221,13 @@ namespace marian {
 
       // this option is only set in the decoder
       if(!lsh_ && options_->hasAndNotEmpty("output-approx-knn")) {
+#ifdef DECODER_ONLY
```

Comment:
LSH is something that's really only useful for decoding.  And we might be replacing lexical shortlists with it in the future.  The `DECODER_ONLY` macro is the wrong one to use here.

Reply:
Got it. I will replace it with `COMPILE_WASM` macro.

@kpu Is there any other concern from your side on this PR? I would like to address all of them with one commit 👍